### PR TITLE
File import/export: add iterator options in the DI

### PR DESCRIPTION
**Pull request by jmleroux**
This PR add the possibilty to configure file iterators from the DI by injecting factory options.

For example, a XLSX file iterator could need read (or skip) a specific worksheet configured in the DI.
If i want options to be injected in my custom file iterator, i need the factory to inject them.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       |no
| Added Behats                      |no
| Changelog updated                 |no
| Review and 2 GTM                  |yes

### CODE REVIEW

File: src/Pim/Component/Connector/Reader/File/Csv/Reader.php [hunk 1]
```diff
@@ -36,14 +36,22 @@ class Reader implements ItemReaderInterface, StepExecutionAwareInterface, Flusha
...
+    protected $options;
+
     /**
      * @param FileIteratorFactory     $fileIteratorFactory
      * @param ArrayConverterInterface $converter
+     * @param array                   $options
      */
-    public function __construct(FileIteratorFactory $fileIteratorFactory, ArrayConverterInterface $converter)
-    {
+    public function __construct(
```
Comment (jjanvier) [inline src/Pim/Component/Connector/Reader/File/Csv/Reader.php]: You should do the same in other readers ;) CSV and xlsx

**Comment by jjanvier** [comment]
GTM once fixed

File: src/Pim/Component/Connector/Reader/File/Csv/Reader.php (hunk 1 above)
Comment (skeleton) [inline src/Pim/Component/Connector/Reader/File/Csv/Reader.php]: yes, don't forger ProductReader & VariantGroup readers :)

**Comment by skeleton** [comment]
gtm once fixed

**Comment by nidup** [comment]
thx @jmleroux !

**Pull request merged**